### DISCUSSION TITLE
FK constraints allowed

### DISCUSSION THREAD

**Discussion by ManolisPap** [planetscale/discussion#158] [Bug Reports]
Hello PlanetScale team,

I have found a bug/issue related to FK constraints. Normally PlanetScale does not support FK constraints. 
But if you run the following sql statements: 

![image](https://user-images.githubusercontent.com/38705760/151434119-1361c18a-60d8-4874-9346-0396467dbe42.png)

The constraints will be added to the schema without an error.
The Line 23 seems to trigger the issue and the constraints are added normally to DB schema without an error.

Result (not expected, the constraints are passed and are showed into DB schema):

![image](https://user-images.githubusercontent.com/38705760/151434170-3994708d-fa7c-4c06-be6c-9f948151f241.png)

On the other hand if line 23(INDEX) was removed PlanetScale gives the following error, which is the expected based on the documentation(of non support of FK constraints)
Error Code: 1317. foreign key constraints are not allowed, see https://vitess.io/blog/2021-06-15-online-ddl-why-no-fk/

![image](https://user-images.githubusercontent.com/38705760/151434208-d56df089-839a-4aef-914f-3654ac7fbdb9.png)

Result (expected):
![image](https://user-images.githubusercontent.com/38705760/151434253-d6dddff5-14cb-4093-a1e7-91c493078608.png)

**Comment by jgreet** [comment]
Thanks! It does appear you can create a foreign key constraint using this syntax. Perhaps the parser doesn't detect this syntax as constraint creation, it's probably something we can fix upstream in Vitess.

PlanetScale's schema linter will catch the constraint if you attempt to promote the branch, or apply the schema change to an existing branch, so this doesn't seem too dangerous.
